src/stim/io/measure_record_reader.test.cc

```diff
@@ -703,3 +708,50 @@ TEST(MeasureRecordReader, FormatDets_InvalidInput) {
     ASSERT_NE(tmp, nullptr);
     ASSERT_THROW({ MeasureRecordReader::make(tmp, SAMPLE_FORMAT_DETS, 3); }, std::runtime_error);
 }
+
+TEST(MeasureRecordReader, read_records_into_RoundTrip) {
+    size_t n_shots = 100;
+    size_t n_results = 512 - 8;
+
+    simd_bit_table shot_maj_data = simd_bit_table::random(n_shots, n_results, SHARED_TEST_RNG());
+    simd_bit_table shot_min_data = shot_maj_data.transposed();
+    for (const auto &kv : format_name_to_enum_map) {
```

Comment:
To clarify: by "change this" I mean I'll rewrite other existing tests to follow the loop-over-formats approach.

Reply:
It's good to have variety in the tests.

Also, this dictionary didn't exist when you wrote the original PR, so it would have been somewhat more inconvenient. The dictionary even includes example python code that is also being tested against the internal code (via the python tests) to ensure consistency across two implementations.